### PR TITLE
feat: Added cookie based user id memorization to support sticky bucketing

### PR DESCRIPTION
**Pull request by zashraf1985**
## Summary
1. Generating a uuid for sticky bucketing.
2. Storage and retrieval of user id from cookie.
3. Added Changelog file to track changes across versions.
4. Updated Readme

### CODE REVIEW

**Comment by zashraf1985** [comment]
@trishahanlon It will be awesome if you can find some time and verify the links here too.

File: src/index.js [hunk 1]
```diff
@@ -14,17 +14,28 @@
...
 const FASTLY_CLIENT_ENGINE = "javascript-sdk/fastly";
+const OPTIMIZELY_USER_ID_COOKIE_NAME = "optimizely_user_id";
 
 addEventListener("fetch", (event) => event.respondWith(handleRequest(event)));
 
 async function handleRequest(event) {
+  const cookies = cookie.parse(event.request.headers.get("Cookie") || '');
+
+  // Fetch user Id from the cookie if available to make sure that a returning user from same browser session always sees the same variation.
+  const userId = cookies[OPTIMIZELY_USER_ID_COOKIE_NAME] || v4();
```
Comment (opti-jnguyen) [inline src/index.js:37]: I think this is fine for now, though one corner case we might need to keep a mental note on is if `v4()` fails for whatever reason in the future (i.e. breaking change to the library).

**Review by opti-jnguyen — APPROVED** [review]
LGTM!

**Comment by trishahanlon** [comment]
@zashraf1985 I'll open a new PR with the Readme link updates. Thanks!

**Comment by zashraf1985** [comment]
@williamoverton You want to take a look?

**Pull request merged**